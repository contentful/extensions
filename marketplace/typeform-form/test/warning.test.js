'use strict';

const React = require('react');
const renderer = require('react-test-renderer');

const { mockComponent } = require('./helpers.js');

jest.mock('@contentful/forma-36-react-components', () => ({
  Icon: mockComponent('icon'),
  Typography: mockComponent('typography'),
  Paragraph: mockComponent('paragraph')
}));

const { Warning } = require('../src/warning.js');

describe('Warning', function() {
  test('renders correctly', function() {
    const tree = renderer.create(<Warning>A warning</Warning>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with different color', function() {
    const tree = renderer
      .create(<Warning iconColor="negative">A more severe warning</Warning>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
