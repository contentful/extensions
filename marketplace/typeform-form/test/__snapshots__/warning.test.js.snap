// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warning renders correctly 1`] = `
<div
  className="typography"
>
  <div
    className="paragraph"
  >
    <div
      className="f36-margin-right--xs"
      color="warning"
      icon="Warning"
    />
    A warning
  </div>
</div>
`;

exports[`Warning renders correctly with different color 1`] = `
<div
  className="typography"
>
  <div
    className="paragraph"
  >
    <div
      className="f36-margin-right--xs"
      color="negative"
      icon="Warning"
    />
    A more severe warning
  </div>
</div>
`;
